testing/mod.ts

```diff
@@ -1,17 +1,72 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 
-import { green, red } from "../colors/mod.ts";
-
+import {
+  bgRed,
+  white,
+  bold,
+  green,
+  red,
+  gray,
+  yellow,
+  italic
+} from "../colors/mod.ts";
 export type TestFunction = () => void | Promise<void>;
 
 export interface TestDefinition {
   fn: TestFunction;
   name: string;
 }
 
+// Replacement of the global `console` function to be in silent mode
+const noop = function(): void {};
+
+// Save Object of the global `console` in case of silent mode
+type Console = typeof window.console;
+// ref https://console.spec.whatwg.org/#console-namespace
+// For historical web-compatibility reasons, the namespace object for
+// console must have as its [[Prototype]] an empty object, created as if
+// by ObjectCreate(%ObjectPrototype%), instead of %ObjectPrototype%.
+const disabledConsole = Object.create({}) as Console;
+Object.assign(disabledConsole, {
+  log: noop,
+  debug: noop,
+  info: noop,
+  dir: noop,
+  warn: noop,
+  error: noop,
+  assert: noop,
+  count: noop,
+  countReset: noop,
+  table: noop,
+  time: noop,
+  timeLog: noop,
+  timeEnd: noop,
+  group: noop,
+  groupCollapsed: noop,
+  groupEnd: noop,
+  clear: noop
+});
+
+const originalConsole = window.console;
+
+function enableConsole(): void {
+  window.console = originalConsole;
+}
+
+function disableConsole(): void {
+  window.console = disabledConsole;
+}
+
+const encoder = new TextEncoder();
+function print(txt: string, carriageReturn: boolean = true): void {
+  if (carriageReturn) {
+    txt += "\n";
```

Comment:
This is not a carriage return character, this a new line character.

s/carriageReturn/newline/g